packages/deepbook-wrapper/sources/helper.move

```diff
@@ -91,6 +93,59 @@ public(package) fun apply_slippage(value: u64, slippage: u64): u64 {
     value + slippage_amount
 }
 
+/// Applies a discount to a value and returns the discounted result
+/// The discount rate is in billionths format (e.g., 50_000_000 = 5%)
+public(package) fun apply_discount(value: u64, discount_rate: u64): u64 {
+    // Verify that the discount is not greater than 100%
+    assert!(discount_rate <= 1_000_000_000, EInvalidDiscountRate);
+
+    let discount_multiplier = 1_000_000_000 - discount_rate;
+    math::mul(value, discount_multiplier)
+}
+
+/// Calculates discount rate based on how much of the DEEP fees the user pays themselves
+/// The more user covers DeepBook fees on their own, the higher the discount rate
+/// If no DEEP fees are required, user gets maximum discount
+public(package) fun calculate_discount_rate(
+    max_discount_rate: u64,
+    deep_from_reserves: u64,
+    deep_required: u64,
+): u64 {
+    assert!(deep_from_reserves <= deep_required, EInvalidDeepFromReserves);
+
+    // If deep_required is 0, give maximum discount
+    if (deep_required == 0) return max_discount_rate;
+
+    let deep_covered_by_user = deep_required - deep_from_reserves;
+
+    // If user covers 0 DEEP, they get 0 discount
+    if (deep_covered_by_user == 0) return 0;
+
+    math::div(math::mul(max_discount_rate, deep_covered_by_user), deep_required)
+}
+
+/// Calculate the taker and maker ratios for an order based on execution status.
+/// Returns (taker_ratio, maker_ratio) in billionths.
+public(package) fun calculate_order_taker_maker_ratio(
+    original_quantity: u64,
+    executed_quantity: u64,
+    order_status: u8,
+): (u64, u64) {
+    // Validate inputs
+    assert!(original_quantity > 0, EZeroOriginalQuantity);
+    assert!(executed_quantity <= original_quantity, EExecutedQuantityExceedsOriginal);
+
+    // An order has maker part only if it's not fully executed and is live or partially filled
```

Comment:
Would be great to extend this description, e.g.:
"...since either order hit the orders that were already in the book and got fulfilled, or partially filled, and the remaining part of the order still sits in the book, which means, if it would be executed, this part would be eligible or maker fees"